tests/test_jsvalidation.py

```diff
@@ -0,0 +1,507 @@
+                ]
+            }
+        }
+    }
+    obj = JSONSchemaValidator()
+    obj.validate(v1, 'first')
+
+
+def test_ref_conflict_inside_schema():
+    """Test that having an invalid reference in the schema raises
+    RefResolutionError."""
+    v1 = {
+        "$schema": "http://json-schema.org/draft-04/schema#",
+
+        "type": "object",
+
+        "properties": {
+            "billing_address": {"$ref": "#/definitions/differentaddress"},
+            "shipping_address": {
+                "allOf": [
+                    {"$ref": "#/definitions/address"},
+                    {
+                        "properties":   {
+                            "type": {"enum": ["residential", "business"]}
+                        }
+                    }
+                ]
+            }
+        }
+    }
+    obj = JSONSchemaValidator()
+    with pytest.raises(jsonschema.exceptions.RefResolutionError):
+        obj.validate(v1, 'first')
+
+
+def test_ref_conflict_between_schemas():
+    """Test that having a conflict between schemas with references passes."""
+    v1 = {
+        "$schema": "http://json-schema.org/draft-04/schema#",
+
+        "definitions": {
+            "address": {
+                "type": "object",
+                "properties": {
+                    "street_address": {"type": "string"}
+                }
+            }
+        },
+
+        "type": "object",
+
+        "properties": {
+            "billing_address": {"$ref": "#/definitions/address"},
+            "shipping_address": {
+                "allOf": [
+                    {"$ref": "#/definitions/address"},
+                    {
+                        "properties":   {
+                            "type": {"enum": ["residential", "business"]}
+                        }
+                    }
+                ]
+            }
+        }
+    }
+    v2 = {
+        "$schema": "http://json-schema.org/draft-04/schema#",
+
+        "definitions": {
+            "address": {
+                "type": "object",
+                "properties": {
+                    "street_address": {"type": "differenttype"}
+                }
+            }
+        },
+
+        "type": "object",
+
+        "properties": {
+            "billing_address": {"$ref": "#/definitions/address"},
+            "shipping_address": {
+                "allOf": [
+                    {"$ref": "#/definitions/address"},
+                    {
+                        "properties":   {
+                            "type": {"enum": ["residential", "business"]}
+                        }
+                    }
+                ]
+            }
+        }
+    }
+    obj = JSONSchemaValidator()
+    obj.validate(v1, 'first')
+    with pytest.raises(JSONSchemaCompatibilityError):
+        obj.validate(v2, 'second')
+
+
+def test_none_types_pass():
+    """FIXME: Later we will want to add an optional failure for unknown types.
+    Test that having a field with no type passes."""
```

Comment:
Could you please format this docstring like that:
```
"""Test that having a field with no type passes.

FIXME: Later we will want to add an optional failure for unknown types.
"""
```
This is just to follow PEP257 ["Multi-line docstrings consist of a summary line just like a one-line docstring, followed by a blank line, followed by a more elaborate description."](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings)